src/main/java/com/adroll/cantor/HLLCounter.java

```diff
@@ -311,11 +311,21 @@ public void combine(HLLCounter h) {
     a = getAlpha(m);
 
     if(intersectable && h.isIntersectable()) {
-      ts.addAll(h.getMinHash());
-      int curSize = ts.size();
       int mink = Math.min(k, h.getK());
-      for(int i = 0; i < (curSize - mink); i++) {
-        ts.pollLast();
+      Iterator<Long> it = h.getMinHash().iterator();
+      if (ts.size() + h.getMinHash().size() < mink) {
+        ts.addAll(h.getMinHash());
+      } else {
```

Comment:
If ts.size() + other HLL's ts.size() is less than mink, then we have enough room in ts to hold all the values. If we don't have enough room, we are going to merge ts of the two HLLs and only keep the bottom mink values. The old code would add all the values from the other ts to this HLL's ts and discard the top (curSize - mink) values. The new code ads values into ts one-by-one until we hit a point where any value added will be discarded anyway, so we can stop